api/v1alpha1/shared_types.go

```diff
@@ -663,7 +663,14 @@ type CustomResponse struct {
 	ContentType *string `json:"contentType,omitempty"`
 
 	// Body of the Custom Response
-	Body CustomResponseBody `json:"body"`
+	//
+	// +optional
+	Body *CustomResponseBody `json:"body,omitempty"`
+
+	// Status Code of the HTTP response
+	//
+	// +optional
+	StatusCode *string `json:"statusCode,omitempty"`
```

Comment:
this is optional? what will happen if both `body` and `statusCode` is nil?

Reply:
Empty status code in this context will be a 200

Can add a cel validation to make sure at least one field must be set

Reply:
> Empty status code for direct response will be a 200.
For response override it is the existing status code

Should address this? maybe not share the struct for better doc(I'm not sure about this)?

Reply:
Yup battling that same thought rn :)

Reply:
Prefer two different types for direct response and override response for below reasons:
* For better doc as @zirain pointed out
* They may have different fields, for example, override response could add/remove headers